tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -733,6 +733,284 @@ async fn handle_native_task<
+						log::error!("Failed to send response");
+					}
+					return;
+				};
+
+				let verify_result = ctx
+					.pumpx_api
+					.verify_google_code(&access_token, google_code.to_string(), language.clone())
+					.await;
+
+				let verify_success = match verify_result {
+					Ok(res) => res.data.result,
+					Err(_) => {
+						log::error!("Google code verification request failed");
+						false
+					},
+				};
+
+				if !verify_success {
+					let response = NativeTaskResponse::Err(NativeTaskError::PumpxApiError(
+						PumpxApiError::GoogleCodeVerificationFailed,
+					));
+					if response_sender.send(response.encode()).is_err() {
+						log::error!("Failed to send response");
+					}
+					return;
+				}
+			}
+
+			// 2. Verify we have a valid Pumpx "access" token for the user
+			let storage = PumpxJwtStorage::new(ctx.storage_db.clone());
+			let Some(access_token) =
+				storage.get(&(sender.to_omni_account(), AUTH_TOKEN_ACCESS_TYPE))
+			else {
+				let response = NativeTaskResponse::Err(NativeTaskError::InternalError);
+				if response_sender.send(response.encode()).is_err() {
+					log::error!("Failed to send response");
+				}
+				return;
+			};
+
+			// 3. Create an unsigned tx with Pumpx backend
+			let create_transfer_res = match ctx
+				.pumpx_api
+				.create_transfer_unsigned_tx(
+					&access_token,
+					request_id,
+					chain_id,
+					wallet_index,
+					&recipient_address,
+					&token_ca,
+					&amount,
+					language.clone(),
+				)
+				.await
+			{
+				Ok(res) => res,
+				Err(e) => {
+					log::error!("Failed to create_transfer_unsigned_tx: {:?}", e);
+					let response = NativeTaskResponse::Err(NativeTaskError::PumpxApiError(
+						PumpxApiError::CreateTransferUnsignedTxFailed,
+					));
+					if response_sender.send(response.encode()).is_err() {
+						log::error!("Failed to send response");
+					}
+					return;
+				},
+			};
+
+			let transfer_id = create_transfer_res.data.transfer_id;
+			let tx_data = create_transfer_res.data.tx_data;
+			// Check if tx_data is present
+			let tx_data = match tx_data {
+				Some(data) => data,
+				None => {
+					log::error!("No tx_data in create_transfer_unsigned_tx response");
+					let response = NativeTaskResponse::Err(NativeTaskError::PumpxApiError(
+						PumpxApiError::CreateTransferUnsignedTxFailed,
+					));
+					if response_sender.send(response.encode()).is_err() {
+						log::error!("Failed to send response");
+					}
+					return;
+				},
+			};
+
+			let Some(chain_type) = ChainType::from_pumpx_chain_id(chain_id) else {
+				log::error!("Failed to map pumpx chain_id {}", chain_id);
+				let response = NativeTaskResponse::Err(NativeTaskError::InternalError);
+				if response_sender.send(response.encode()).is_err() {
+					log::error!("Failed to send response");
+				}
+				return;
+			};
+
+			// 4. Use the pumpx_signer_client to sign each tx_data entry in parallel
+			let mut sign_futures = Vec::new();
+			for tx in tx_data {
```

Comment:
we should probably extract this logic to some service we can reuse as this will be used in multiple places. But it can be done in a separate PR

Reply:
refactored to use "request_signatures"